### PR TITLE
fix: resolve slip44 key mismatch for native tokens in assets-unify-state selectors

### PR DESCRIPTION
**Pull request by salimtb**
## **Description**

When assets-unify-state is enabled, assetsInfo stores native EVM tokens using slip44:60 (EVM generic coin type) while assetsPrice may use a chain-specific slip44 (e.g. slip44:9005 for AVAX on Avalanche C-Chain).

This mismatch caused two selectors to silently skip affected native tokens:
- getCurrencyRateControllerCurrencyRates: no conversion rate for AVAX, making token.secondary undefined on the token list
- getTokenRatesControllerMarketData: no market data for AVAX, making pricePercentChange1d undefined and hiding the % change badge

Both selectors now fall back to a chain-prefix scan when the direct assetId lookup misses, bridging the key mismatch without changing stored state.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/40964?quickstart=1)

## **Changelog**

CHANGELOG entry: resolve slip44 key mismatch for native tokens in assets-unify-state selectors

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

### **After**

https://github.com/user-attachments/assets/652950ce-2af4-4d78-9445-22bc08be1a17

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

---

> [!NOTE]
> **Medium Risk**
> Touches rate/market-data selectors used for token pricing and currency conversion; the new chain-prefix fallback could select an unintended slip44 entry if multiple native-price keys exist for a chain.
> 
> **Overview**
> Fixes assets-unify-state selectors to handle native EVM tokens when `assetsInfo` uses `slip44:60` but `assetsPrice` uses a chain-specific `slip44` (e.g. AVAX). `getCurrencyRateControllerCurrencyRates` now falls back to any native `slip44:*` price entry for the same chain when a direct lookup misses.
> 
> Similarly, `getTokenRatesControllerMarketData` adds a guarded fallback (only for native `slip44` assets) to find matching native metadata in `assetsInfo`, preventing misattribution of ERC-20 prices.
> 
> Expands `assets-migration.test.ts` with many new edge-case tests across the migrated selectors (deduping `customAssets`, skipping missing metadata/non-EVM assets, multi-account/multi-chain behavior, decimal parsing), including explicit coverage for the slip44 mismatch scenarios.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit ef0e39ed56046f3fc428f4144fc2d3e209acd67e. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: shared/lib/selectors/assets-migration.ts [hunk 1]
```diff
@@ -659,17 +659,33 @@ export const getCurrencyRateControllerCurrencyRates = createDeepEqualSelector(
...
       if (
         metadata.type !== 'native' ||
-        assetType.chain.namespace !== KnownCaipNamespace.Eip155 ||
-        !price
+        assetType.chain.namespace !== KnownCaipNamespace.Eip155
       ) {
         continue;
       }
 
+      // assetsInfo may use slip44:60 (EVM standard) for native tokens while assetsPrice
```
Comment (bergarces) [inline shared/lib/selectors/assets-migration.ts:671]: Is there something we can do in the future to stop assetsInfo having the wrong assetId?

**Review by bergarces — APPROVED** [review]

File: shared/lib/selectors/assets-migration.ts (hunk 1 above)
Comment (salimtb) [reply shared/lib/selectors/assets-migration.ts:671]: we can hardcode a list but this is not the best approach

**Review by Prithpal-Sooriya — APPROVED** [review]
